### ISSUE TITLE
Are you open to submitting Alltin-DIN to Google Fonts?

### ISSUE THREAD

**Issue by driehuis** [Altinn/altinn-din#1]
There are no DIN-style fonts in Google Fonts, as far as I can tell. I would love to see Altinn-DIN be added. I'm not associated with Google in any other way than being an end-user, and IANAL, but I see very few obstacles on the road there. If you're interested, please review https://github.com/google/fonts/blob/main/CONTRIBUTING.md for the details.

Of the chief requirements mentioned there, Altinn-DIN ticks most of the boxes. The font has to be OFL licensed☑, should have a public git repository☑, have open participation☑, and meet quality standards (looking good). A quick review shows that the changes would be fairly limited:

* There must be no Reserved Font Name in the license
* I ran the FontBakery test on Altinn-DIN's Regular OTF, and it came with a few minor tweaks (and one of them will actually result in the font becoming more consistent between Mac/Windows/Linux and assorted software, the rest is dotting the i's and crossing the t's if you pardon the pun).

So, the questions I have before trying to fix the technical nits are:

* Would you be willing to drop the Reserved Font Names?
* Would you be willing to rename the font? Google appears to have infrastructure within Google Fonts to credit Alltin in other ways.
* Are you accepting pull requests to tackle the minor issues uncovered by fontbakery?

I could easily resolve these issues by forking the repo and take the _brand of footwear_ approach and Just Do _Something_, but I know I'm a far better contributor than I am an organizer, and I'm not a big fan of the eternal forking of open source, so I thought I'd ask!

**Comment by wayward-turnip** [comment]
FWIW, Great idea to add this to Google Fonts.  As a prosumer-type-end-user, I would like to have this avail via the G.Fonts infrastructure, (but I'm not sure what the differences (if any) are between Altinn and D-DIN).

**Comment by driehuis** [comment]
As it turned out, making the Google QA software happy with it turned out to be a significant chunk of work. Font names with dashes turned out to be off limits, as are font names with companies' names in it. I wound up forking this repo, making a ton of changes, and in the end submitted it to Google Fonts as the "DINish" font family. It currently sits in the queue as https://github.com/google/fonts/issues/3377. Unfortunately, Google appears to be swamped with font submissions so all I can do for now is wait...

**Comment by altinnadmin** [comment]
Hi @driehuis!
SO sorry for not noticing this issue before now!! 😞 

I thought nobody out there would notice this little repo, and we've been super-busy working on the open source [next gen Altinn](https://github.com/Altinn/altinn-studio#altinn3) this last year. Note to self: Always watch repos I create :-P

We definitely take pull requests, so if you'd like to contribute your changes back, we'd love that.

Collaborating on this font with someone who *actually* knows what they're doing would be great 😅, and now that Altinn has become a [digital public good](https://digitalpublicgoods.net/registry/), it could be a good thing for so many more people around the world.

For example, what are your thoughts on #2 ?

Also, if naming is a problem getting this font "out there", we will consider changing the name.
Internally we've been joking that "altdin" would be a much better, shorter and funnier name anyway ;)

**Comment by driehuis** [comment]
Sorry for taking a week myself to answer! When I asked, I had a course of action plotted for "yes" and for "no", but I had no plan for no response. :-)

Looking back at this adventure, a lot of things have happened that I did not foresee, and that makes syncing our clocks a challenge. In the process, the fonts have been converted to UFO format, the build tools have been rewritten, and I had to pick a new name out of a hat (this became "DINish"), and register a font foundry ("playbeing", which now has been assigned the OpenType code "PBNG"). As it turned out, making fontbakery happy was a much more involved effort than I had expected. I had to made tons of changes just to placate fontbakery. Notably, the directory structure had to be changed several times. If you look at https://github.com/playbeing/dinish you will probably have a hard time recognizing the original repo. The build infrastructure has changed as well, requiring Python 3 and gmake. This is all within the realm of what I had been expecting as a toolsmith, except more work.

Then the font family got extended with a Condensed Italic. Tabular numbers, old style numbers, and tabular old style numbers have been added, with OpenType support. Your monospaced number requirement has been taken care of already! There is also OpenType magic for Polish and Romanian. All European languages that use the Latin alphabet are now fully supported. Many small inconsistencies were resolved. I created a mini website for DINish, and that lives in the repo too. The project got a bit out of hand :-)

I'm not sure what the best way forward is. I'd be happy to create a pull request, but that would make you inherit the rebranding. You're more than welcome to then rebrand the font "Altdin" (which ticks all the boxes for Google font naming BTW), and register Altinn as a foundry, or even to rebrand it back to Altinn-DIN.

However, at this stage, I think I would prefer to guide DINish through its first steps myself. For me, this is purely an outlet for my creative side, so I have nothing to gain or lose either way. Back when I started this, I thought I'd just be doing some tooling and mechanical changes, and I had never envisioned that I would contribute more glyphs than the original designer! Adding the first new glyph was scary. I think you know the feeling.

Anyway, I do see the value of working together with a group of motivated people with far better skills in everything Web and graphics than I ever will have. I just don't have a picture yet of what the cooperation would look like!

It would probably be a good idea to review the DINish repo first, to see how you would like to proceed.

On the wish list are:
* Adding missing Vietnamese and African Latin glyphs, and the remaining glyphs Google requires for new fonts
* Make spacing of the Bold more consistent (it's not bad, but it could be better)
* Adding Cyrillic and Greek glyphs
* Improving the web site
* Adding more weights, or even converting the font to a variable font

That's a ton of work, I know! Of these items, the extra weights one is the hardest. There are derivatives of D-DIN around that have more weights, but the one I looked at was quite unpolished in the kerning department.

**Comment by altinnadmin** [comment]
You've certainly been busy @driehuis, impressive! This is why we just *love* doing open source.

It seems like you have, in addition to all the cleanup, made some key improvements that are important to us:

1. Tabular ("monospace") numbers 👌 
2. Additional glyphs/languages.
    Question: Does this mean that the special characters used in [sami](https://en.wikipedia.org/wiki/S%C3%A1mi_orthography#The_eight_literary_languages) is also supported now? That would be fantastic.

From our perspective it makes sense to bring your changes back instead of redoing the same work again.
I'll have a look at the changes you've done in the repo to get a feel for how that can be done, and I'll also do some thinking about how we can cooperate.

Having a variable version of this font would be a wet dream come true, and in addition to the other wishes, could make this a very popular font indeed.

**Comment by driehuis** [comment]
Hmmm, Sámi has a ton of different orthographies, and was flying under my radar because it only has 25,000 speakers and my cutoff was set at 100,000. I hadn't realised that. I presume you are looking at the Northern Sámi orthography? That is missing four glyphs. I will add those shortly (probably in a few weeks, as I'm quite busy at the moment).

A variable font is quite a long way away, I'm afraid. Once we have a 900 and a 100 weight the heavy lifting is probably done, but the fun just starts there. The OpenType magic required will likely be extensive. As I mentioned, I'm not happy with the spacing of the current bold, and that's just one weight. The heavier version of D-DIN that's on Github somewhere requires tons of fixes, and at that stage will still not have the required glyphs. I have never even seen an attempt at a lighter than 400 weight.

**Comment by altinnadmin** [comment]
>  I presume you are looking at the Northern Sámi orthography? That is missing four glyphs.

Correct.

> I will add those shortly (probably in a few weeks, as I'm quite busy at the moment).

Excellent! 👏 

> A variable font is quite a long way away

Yeah, that's why it's only a dream... :-P But looking ahead, that's the way "the industry" seems to be going.

**Comment by driehuis** [comment]
I've sketched up the missing characters for Northern Sámi. Could you download Dinish-Regular from https://github.com/playbeing/dinish/tree/add-northern-sami/fonts/ttf/Dinish and review the new glyphs?

In particular for the capital Eng, I saw two radically different shapes in examples of Sami orthography.

**Comment by altinnadmin** [comment]
Thanks, I'm rather busy today, but this link describes the Sámi characters that are mandatory. Sorry, page only in Norwegian... :(
https://www.digdir.no/felleslosninger/teiknsett/1498

The characters:
Č, č, Đ, đ, Ŋ, ŋ, Š, š, Ŧ, ŧ, Ž, ž

Screenshot (font used is [Inter](https://fonts.google.com/specimen/Inter)):
![image](https://user-images.githubusercontent.com/6088624/137872546-ab1ffccf-04ad-4c34-a752-41a4044c73d8.png)

**Comment by lvbachmann** [comment]
> In particular for the capital Eng, I saw two radically different shapes in examples of Sami orthography.

One of the main reasons you won't find much consistency here is because this letter (AFAIK) never appears as the first letter of any word or name in Sámi language. Thus the need to standardise is lower than with glyphs that are more commonly/naturally in use. As for our use we try to avoid texts in all caps, so we probably won't need a capital eŋ at any point.

**Comment by driehuis** [comment]
Excellent, thanks for the feedback! I noticed the "đ Latin Small Letter D with Stroke" needs correcting, I will make a note of that.

**Comment by slk333** [comment]
Would be cool to have it on Google Fonts

**Comment by altinnadmin** [comment]
> Would be cool to have it on Google Fonts

@slk333 Yup, but even cooler would be to have [DINish](https://github.com/playbeing/dinish) there :)

It's a much improved and also actively maintained fork of Altinn-DIN.
